pkg/deploy/controller/deploymentconfig/factory.go

```diff
@@ -81,3 +81,259 @@ func (factory *DeploymentConfigControllerFactory) Create() controller.RunnableCo
+		if !ok {
+			glog.Errorf("Tombstone contained object that is not a deployment config: %+v", obj)
+			return
+		}
+	}
+	glog.V(4).Infof("Deleting deployment config %q", dc.Name)
+}
+
+// addReplicationController enqueues the deployment that manages a replicationcontroller when the replicationcontroller is created.
+func (c *DeploymentConfigController) addReplicationController(obj interface{}) {
+	rc := obj.(*kapi.ReplicationController)
+	glog.V(4).Infof("Replication controller %q added.", rc.Name)
+	if dc, err := c.dcStore.GetConfigForController(rc); err == nil && dc != nil {
+		c.enqueueDeploymentConfig(dc)
+	}
+}
+
+// updateReplicationController figures out which deploymentconfig is managing this replication
+// controller and requeues the deployment config.
+func (c *DeploymentConfigController) updateReplicationController(old, cur interface{}) {
+	// A periodic relist will send update events for all known controllers.
+	if kapi.Semantic.DeepEqual(old, cur) {
+		return
+	}
+
+	curRC := cur.(*kapi.ReplicationController)
+	glog.V(4).Infof("Replication controller %q updated.", curRC.Name)
+	if dc, err := c.dcStore.GetConfigForController(curRC); err == nil && dc != nil {
+		c.enqueueDeploymentConfig(dc)
+	}
+}
+
+// deleteReplicationController enqueues the deployment that manages a replicationcontroller when
+// the replicationcontroller is deleted. obj could be an *kapi.ReplicationController, or
+// a DeletionFinalStateUnknown marker item.
+func (c *DeploymentConfigController) deleteReplicationController(obj interface{}) {
+	rc, ok := obj.(*kapi.ReplicationController)
+
+	// When a delete is dropped, the relist will notice a pod in the store not
+	// in the list, leading to the insertion of a tombstone object which contains
+	// the deleted key/value.
+	if !ok {
+		tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
+		if !ok {
+			glog.Errorf("Couldn't get object from tombstone %+v, could take up to %v before a deployment config recreates/updates replicationcontrollers", obj, FullDeploymentConfigResyncPeriod)
+			return
+		}
+		rc, ok = tombstone.Obj.(*kapi.ReplicationController)
+		if !ok {
+			glog.Errorf("Tombstone contained object that is not a replication controller %+v, could take up to %v before a deployment recreates/updates replicationcontrollers", obj, FullDeploymentConfigResyncPeriod)
+			return
+		}
+	}
+	glog.V(4).Infof("Replication controller %q deleted.", rc.Name)
+	if dc, err := c.dcStore.GetConfigForController(rc); err == nil && dc != nil {
+		c.enqueueDeploymentConfig(dc)
+	}
+}
+
+func (c *DeploymentConfigController) enqueueDeploymentConfig(dc *deployapi.DeploymentConfig) {
+	key, err := kcontroller.KeyFunc(dc)
+	if err != nil {
+		glog.Errorf("Couldn't get key for object %+v: %v", dc, err)
+		return
+	}
+	c.queue.Add(key)
+}
+
+// worker runs a worker thread that just dequeues items, processes them, and marks them done.
+func (c *DeploymentConfigController) worker() {
+	for {
+		func() {
```

Comment:
I'm not sure why the for loop even exists here: the function of wait.Until is to loop/jitter for you, right?